### PR TITLE
Renames "Cyborg Analyzer" to "Machine Analyzer", it now scans how damaged machines are too!

### PR DESCRIPTION
**Pull request by Fordoxia**
## What Does This PR Do
Renames the cyborg analyser, gives it the ability to scan machines to see how much damage they have sustained.  Slightly alters the scan dialog for scanning synthetics.

(Also updates the comments a lil)

## Why It's Good For The Game
You can now identify more-or-less exactly how much damage a machine has sustained, giving the analyser a little bit of extra use.

(updated comments good for code-reading)

## Testing
Spawned in. Scanned a bunch of machines. Proceeded to hit and blast the machines with dangerous weapons. Scanned the machines again. Scanner worked fine and told me how much integrity the machines had left.

## Changelog
:cl:
add: Newly renamed Machine analyser can now scan machines in addition to crew and borgs.
tweak: Cyborg analyzer renamed to Machine analyzer. Slightly alters the scan dialog for scanning synthetics.
/:cl:

### CODE REVIEW

File: code/game/machinery/machinery.dm [hunk 1]
```diff
@@ -348,6 +348,22 @@
...
+			return
+		if(stat & BROKEN)
+			to_chat(user, "<span class='warning'>Catastrophic structural damage detected! [src] requires complete reconstruction.</span>")
+			return
+		else
+			to_chat(user, "<span class='info'>Structural damage detected! [src]'s estimated integraty is [round((obj_integrity/max_integrity)*100)]%.</span>")
+			return
+	else
+		return ..()
+
```
Comment (CinnamonSnowball) [inline code/game/machinery/machinery.dm:366]: This should probably be in the machine analyzer's `attack()` instead
Currently, that proc only assumes input of a living mob and runs `robot_healthscan()` on it, but we could instead make it accept any atom and run `robot_healthscan()`/a new `machine_scan()` or sth, according to what was smacked

Comment (lewcc) [reply code/game/machinery/machinery.dm:366]: Yeah, the brunt of this behavior should probably be a function of the analyzer more than the machine

File: code/game/objects/items/devices/scanners.dm [hunk 2]
```diff
@@ -354,7 +367,35 @@ REAGENT SCANNER
...
+	if(M.obj_integrity == M.max_integrity)
+		to_chat(user, "<span class='info'>[M] is at full integraty.</span>")
+		return
+	else
+		to_chat(user, "<span class='info'>Structural damage detected! [M]'s overall estimated integraty is [round((M.obj_integrity/M.max_integrity)*100)]%.</span>")
+		if(M.stat & BROKEN) // Displays alongside above message. Machines with a "broken" state do not become broken at 0% HP - anything that reaches that point is destroyed
+			to_chat(user, "<span class='warning'>Further analysis: Catastrophic component failure detected! [M] requires reconstruction to fully repair.</span>")
+			return
+		return
+
```
Comment (DGamerL) [inline code/game/objects/items/devices/scanners.dm:384]:
```suggestion
```

Comment (DGamerL) [inline code/game/objects/items/devices/scanners.dm:384]:
```suggestion
```

File: code/game/objects/items/devices/scanners.dm [hunk 3]
```diff
@@ -354,7 +367,35 @@ REAGENT SCANNER
...
+	add_fingerprint(user)
+
+/proc/machine_scan(mob/user, obj/machinery/M)
+	if(M.obj_integrity == M.max_integrity)
+		to_chat(user, "<span class='info'>[M] is at full integraty.</span>")
+		return
+	else
+		to_chat(user, "<span class='info'>Structural damage detected! [M]'s overall estimated integraty is [round((M.obj_integrity/M.max_integrity)*100)]%.</span>")
+		if(M.stat & BROKEN) // Displays alongside above message. Machines with a "broken" state do not become broken at 0% HP - anything that reaches that point is destroyed
+			to_chat(user, "<span class='warning'>Further analysis: Catastrophic component failure detected! [M] requires reconstruction to fully repair.</span>")
```
Comment (DGamerL) [inline code/game/objects/items/devices/scanners.dm:381]:
```suggestion
	if(M.obj_integrity == M.max_integrity)
		to_chat(user, "<span class='info'>[M] is at full integrity.</span>")
		return
	to_chat(user, "<span class='info'>Structural damage detected! [M]'s overall estimated integrity is [round((M.obj_integrity / M.max_integrity) * 100)]%.</span>")
	if(M.stat & BROKEN) // Displays alongside above message. Machines with a "broken" state do not become broken at 0% HP - anything that reaches that point is destroyed
		to_chat(user, "<span class='warning'>Further analysis: Catastrophic component failure detected! [M] requires reconstruction to fully repair.</span>")
```

File: code/game/objects/items/devices/scanners.dm [hunk 4]
```diff
@@ -343,7 +356,30 @@ REAGENT SCANNER
...
+/obj/item/robotanalyzer/attack_obj(obj/machinery/M, mob/living/user) // Scanning a machine object
+	if((HAS_TRAIT(user, TRAIT_CLUMSY) || user.getBrainLoss() >= 60) && prob(50))
+		var/list/msgs = list()
+		user.visible_message("<span class='warning'>[user] has analyzed the floor's components!</span>", "<span class='warning'>You try to analyze the floor's vitals!</span>")
+		msgs += "<span class='info'>Analyzing Results for The floor:\n\t Overall Status: Unknown</span>"
+		msgs += "<span class='info'>\t Damage Specifics: <font color='#FFA500'>[0]</font>/<font color='red>[0]</font></span>"
+		msgs += "<span class='info'>Key: <font color='#FFA500'>Burns</font><font color ='red'>/Brute</font></span>"
+		msgs += "<span class='info'>Chassis Temperature: ???</span>"
+		to_chat(user, chat_box_healthscan(msgs.Join("<br>")))
+		return
```
Comment (DGamerL) [inline code/game/objects/items/devices/scanners.dm:368]: Could you make this into it's own helper proc?

File: code/game/objects/items/devices/scanners.dm [hunk 5]
```diff
@@ -343,18 +356,36 @@ REAGENT SCANNER
...
+	to_chat(user, "<span class='info'>Structural damage detected! [M]'s overall estimated integrity is [round((M.obj_integrity / M.max_integrity) * 100)]%.</span>")
+	if(M.stat & BROKEN) // Displays alongside above message. Machines with a "broken" state do not become broken at 0% HP - anything that reaches that point is destroyed
+		to_chat(user, "<span class='warning'>Further analysis: Catastrophic component failure detected! [M] requires reconstruction to fully repair.</span>")
 
-	user.visible_message("<span class='notice'>[user] has analyzed [M]'s components.</span>", "<span class='notice'>You have analyzed [M]'s components.</span>")
+/obj/item/robotanalyzer/attack(mob/living/M, mob/living/user) // Scanning borgs, IPCs/augmented crew, and AIs
+	if((HAS_TRAIT(user, TRAIT_CLUMSY) || user.getBrainLoss() >= 60) && prob(50))
+		handle_clumsy(user)
+		return
+	user.visible_message("<span class='notice'>[user] has analyzed [M]'s components with the [src].</span>", "<span class='notice'>You analyze [M]'s components with the [src].</span>")
```
Comment (lewcc) [inline code/game/objects/items/devices/scanners.dm:388]:
```suggestion
	user.visible_message("<span class='notice'>[user] has analyzed [M]'s components with [src].</span>", "<span class='notice'>You analyze [M]'s components with [src].</span>")
```

File: code/game/objects/items/devices/scanners.dm [hunk 6]
```diff
@@ -343,18 +356,36 @@ REAGENT SCANNER
...
-		var/list/msgs = list()
-		user.visible_message("<span class='warning'>[user] has analyzed the floor's components!</span>", "<span class='warning'>You try to analyze the floor's vitals!</span>")
-		msgs += "<span class='info'>Analyzing Results for The floor:\n\t Overall Status: Unknown</span>"
-		msgs += "<span class='info'>\t Damage Specifics: <font color='#FFA500'>[0]</font>/<font color='red>[0]</font></span>"
-		msgs += "<span class='info'>Key: <font color='#FFA500'>Burns</font><font color ='red'>/Brute</font></span>"
-		msgs += "<span class='info'>Chassis Temperature: ???</span>"
-		to_chat(user, chat_box_healthscan(msgs.Join("<br>")))
+		handle_clumsy(user)
+		return
+	user.visible_message("<span class='notice'>[user] has analyzed [M]'s components with the [src].</span>", "<span class='notice'>You analyze [M]'s components with the [src].</span>")
```
Comment (lewcc) [inline code/game/objects/items/devices/scanners.dm:372]: `the` is automatically added if needed when referring to objects like this
```suggestion
	user.visible_message("<span class='notice'>[user] has analyzed [M]'s components with [src].</span>", "<span class='notice'>You analyze [M]'s components with [src].</span>")
```

**Review by lewcc — COMMENTED** [review]
Just some little things. I'm assuming you also tested this on borgs after the changes to verify it still works on them?

**Comment by Fordoxia** [comment]
> Just some little things. I'm assuming you also tested this on borgs after the changes to verify it still works on them?

Yes, I tested it on borgs, IPCs, and people with prosthetics and that all still works

**Review by Henri215 — CHANGES_REQUESTED** [review]
There are other places in the code where the "cyborg analyzer" needs to be renamed (medical_designs for example).

**Comment by Fordoxia** [comment]
> There are other places in the code where the "cyborg analyzer" needs to be renamed (medical_designs for example).

I did the thing.

File: code/modules/research/designs/medical_designs.dm [hunk 7]
```diff
@@ -33,8 +33,8 @@
 	category = list("Medical")
 
 /datum/design/cyborg_analyzer
```
Comment (Henri215) [inline code/modules/research/designs/medical_designs.dm:35]:
```suggestion
/datum/design/machine_analyzer
```

File: code/modules/research/designs/medical_designs.dm [hunk 8]
```diff
@@ -33,8 +33,8 @@
...
-	name = "Cyborg Analyzer"
-	desc = "A hand-held scanner able to diagnose robotic injuries."
+	name = "Machine Analyzer"
+	desc = "A hand-held scanner able to diagnose robotic injuries and the condition of machinery."
 	id = "cyborg_analyzer"
```
Comment (Henri215) [inline code/modules/research/designs/medical_designs.dm:38]:
```suggestion
	id = "machine_analyzer"
```

File: code/game/objects/items/devices/scanners.dm [hunk 9]
```diff
@@ -343,18 +356,36 @@ REAGENT SCANNER
...
-		msgs += "<span class='info'>Key: <font color='#FFA500'>Burns</font><font color ='red'>/Brute</font></span>"
-		msgs += "<span class='info'>Chassis Temperature: ???</span>"
-		to_chat(user, chat_box_healthscan(msgs.Join("<br>")))
+		handle_clumsy(user)
+		return
+	user.visible_message("<span class='notice'>[user] has analyzed [M]'s components with [src].</span>", "<span class='notice'>You analyze [M]'s components with [src].</span>")
+	machine_scan(user, M)
+	add_fingerprint(user)
+
+/proc/machine_scan(mob/user, obj/machinery/M)
```
Comment (Henri215) [inline code/game/objects/items/devices/scanners.dm:376]: Missing the whole path here?

Comment (Fordoxia) [reply code/game/objects/items/devices/scanners.dm:376]: What is missing?

Comment (Henri215) [reply code/game/objects/items/devices/scanners.dm:376]: ```/obj/item/robotanalyzer/proc/machine_scan(mob/user, obj/machinery/M)```

Comment (Fordoxia) [reply code/game/objects/items/devices/scanners.dm]: I am very smart. Fixed.

**Review by Henri215 — APPROVED** [review]
lewcs good to me

**Review by S34NW — APPROVED** [review]

**Pull request merged**